pyteal/ast/subroutine.py

```diff
@@ -49,15 +55,32 @@ def __init__(
                     )
                 )
 
+            expected_arg_type: type = Expr
```

Comment:
@tzaffi Can you help me understand how it's safe to assume the type is `Expr` here?

* It appears a check for `Expr` happens _after_ building `expected_arg_types` on line 76.
* My interpretation is:  It's _not_ safe to assume the data type is `Expr` here.  Though any violating data types will be caught on line 76.
* If my understanding is accurate, then:
  * My concern is someone will later change ordering/validation logic that unexpectedly retains an undesired data type.
  * I'd try to perform validation _before_ (or at the same time as) building `expected_arg_types`.

Reply:
👍